cypress/integration/contact/contact.test.js

```diff
@@ -1,65 +1,144 @@
-    cy.get('[data-cy=uaSchedule]').should('have.text', uaSchedule);
-    cy.get('[data-cy=enSchedule]').should('have.text', enSchedule);
-    cy.get('[data-cy=uaAddress]').should('have.text', uaAddress);
-    cy.get('[data-cy=enAddress]').should('have.text', enAddress);
-    cy.get('[data-cy=uaCartImage]')
-      .should('have.attr', 'src')
-      .should('include', uaCartImage);
-    cy.get('[data-cy=enCartImage]')
-      .should('have.attr', 'src')
-      .should('include', enCartImage);
-    cy.get('[data-cy=mapLink]').should('have.text', cartLink);
-    cy.get('[data-cy=email]').should('have.text', email);
-    cy.get('[data-cy=goBackButton]')
-      .should('be.visible')
-      .should('have.text', 'Назад')
-      .click()
-      .wait(2000);
+    cy.get('@table').eq(3).children().first().click();
+    cy.wait(1000);
+    cy.get('[data-cy=phone-number]').should('be.visible');
+    cy.get('[data-cy=ua-schedule]').should('be.visible');
+    cy.get('[data-cy=en-schedule]').should('be.visible');
+    cy.get('[data-cy=ua-address]').should('be.visible');
+    cy.get('[data-cy=en-address]').should('be.visible');
+    cy.get('[data-cy=map-link]')
+      .children()
+      .next()
+      .children()
+      .eq(0)
+      .should('be.visible');
+    cy.get('[data-cy=email]')
+      .children()
+      .next()
+      .children()
+      .eq(0)
+      .should('be.visible');
+    cy.get('[data-cy=go-back-button]').should('be.visible').click().wait(2000);
     cy.location('pathname', { timeout: 1000 }).should('eq', '/contacts');
   });
+  it('The data of new contact should be add', () => {
+    cy.stubRequest('addContacts', addContacts).as('addContacts');
+    cy.get('[data-cy=add-contact]').click();
+    cy.wait(1000);
+    cy.fixture('HORONDI.png').then((fileContent) => {
+      cy.get('[data-cy=upload-photo]').attachFile({
+        fileContent: fileContent.toString(),
+        mimeType: 'image/png',
+        filePath: '../fixtures/HORONDI.png'
+      });
+    });
+    cy.fixture('HORONDI.png').then((fileContent) => {
+      cy.get('[data-cy=upload-photo]').attachFile({
+        fileContent: fileContent.toString(),
+        mimeType: 'image/png',
+        filePath: '../fixtures/HORONDI.png'
+      });
+    });
+    cy.get('[data-cy=map-link]').type(contactToAdd[0].mapLink);
+    cy.get('[data-cy=phone-number]').type(contactToAdd[0].contactNumber);
+    cy.get('[data-cy=ua-schedule]').type(contactToAdd[0].scheduleUa);
+    cy.get('[data-cy=en-schedule]').type(contactToAdd[0].scheduleEn);
+    cy.get('[data-cy=ua-address]').type(contactToAdd[0].addressUa);
+    cy.get('[data-cy=en-address]').type(contactToAdd[0].addressEn);
+    cy.get('[ data-cy=email]').type(contactToAdd[0].email);
+    cy.get('[ data-cy=save]').click();
+    cy.wait(3000);
+    cy.get('.MuiAlert-message').should('be.visible');
+  });
+  it('Tne data should be changed', () => {
+    cy.stubRequest('updateContact', updateContact).as('updateContact');
```

Comment:
We also need to stub request `getContactById`. When we click on click on edit button we sent this request for server.